components/payment_form/address_form.tsx

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState} from 'react';
+
+import {FormattedMessage, MessageDescriptor, useIntl} from 'react-intl';
+
+import {Address} from '@mattermost/types/cloud';
+import DropdownInput from 'components/dropdown_input';
+import {COUNTRIES} from 'utils/countries';
+import Input from 'components/widgets/inputs/input/input';
+
+import StateSelector from './state_selector';
+
+import './payment_form.scss';
+
+type AddressFormProps = {
+    onAddressChange: (address: Address) => void;
+    onBlur: () => void;
+    title: MessageDescriptor;
+    formId: string;
+    initialAddress: Address;
+}
+
+const AddressForm = (props: AddressFormProps) => {
+    const {formatMessage} = useIntl();
+    const [address, setAddress] = useState<Address>(props.initialAddress);
```

Comment:
I've seen this pattern elsewhere, especially, in CWS, of taking in initial state from the parent, making a local copy, maintaining the copy, and sending changes out to the parent. It seems indirect and hard to tell what the source of truth is to me.

This works as is so its not a blocking change, but I think it would be simpler and easier to understand to pass in the parent state and always read it, and rely on the parent handler to update the state that will be sent back in. We always send in part of the parent state already in the form of the partial `{country: ...}` object. Always reading the parent would help us be more meticulous in the parent about sending in a fully fleshed out object instead of a type coerced object, and would let us avoid the `| null` in the parent to begin with.

Reply:
I've updated the whole component to make it controlled. Did it the way it was initially because I basically just pulled the form out of the `payment_form` and adjusted it